src/py/flwr/common/metricsrecord.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""MetricsRecord."""
+
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Union, get_args
+
+from .typing import MetricsScalar, MetricsScalarList
+
+MetricsRecordValues = Union[MetricsScalar, MetricsScalarList]
+
+
+@dataclass
+class MetricsRecord:
+    """Metrics record."""
+
+    data: Dict[str, MetricsRecordValues] = field(default_factory=dict)
```

Comment:
Should we support `keep_input` to improve consistency between `MetricsRecord` and `ParametersRecord`?
```suggestion

    keep_input: bool
    data: Dict[str, MetricsRecordValues] = field(default_factory=dict)
```